### ISSUE TITLE
Add BigQuery support for CREATE TABLE from a SELECT

### ISSUE THREAD

**Issue by zineddine6** [jOOQ/jOOQ#17021]
### Use case

I need to create a table from a select in BigQuery (https://www.jooq.org/doc/latest/manual/sql-building/ddl-statements/create-statement/create-table-statement/create-table-as-select/)

### Possible solution

`CREATE TABLE dataset.my_table AS my_table_alias FROM (SELECT * FROM dataset.my_source);`

### jOOQ Version

3.20-SNAPSHOT

### Database product and version

Latest BigQuery

### Java Version

openJDK 17.0.8.1 2023-08-24

**Comment by lukaseder** [comment]
Thanks a lot for your suggestion. I'll look into this for jOOQ 3.20.0

**Comment by lukaseder** [comment]
Although it doesn't seem to be documented, it seems better to wrap the query in parentheses, i.e. instead of:

```sql
create table TEST.`t`
as
select 'value' as `value`
```

We should generate

```sql
create table TEST.`t`
as (
  select 'value' as `value`
)
```

The former seems to produce errors like this via JDBC (though it works fine in the cloud console):

```
org.jooq.exception.DataAccessException: SQL [create table `t` as select 'value' as `value`]; [Simba][BigQueryJDBCDriver](100032) Error executing query job. Message: BIGQUERY_API_ERR
	at org.jooq_3.20.0-SNAPSHOT.BIGQUERY.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3649)
	at org.jooq.impl.Tools.translate(Tools.java:3622)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:827)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.test.all.testcases.DDLTests.testCreateTableAsSelect(DDLTests.java:5475)
	at org.jooq.test.jOOQAbstractTest.testCreateTableAsSelect(jOOQAbstractTest.java:4718)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:757)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.sql.SQLException: [Simba][BigQueryJDBCDriver](100032) Error executing query job. Message: BIGQUERY_API_ERR
	at com.simba.googlebigquery.googlebigquery.client.requests.jobs.JobsQueryRequest.throwException(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.requests.AbstractRequestWithRetry.executeWithRetry(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.queryclient.JobsQueryClient.executeQuery(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.BQClient.executeQuery(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.dataengine.BQAbstractExecutor.execute(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.dataengine.BQSQLExecutor.execute(Unknown Source)
	at com.simba.googlebigquery.jdbc.common.SStatement.executeNoParams(Unknown Source)
	at com.simba.googlebigquery.jdbc.common.BaseStatement.execute(Unknown Source)
	at org.jooq.tools.jdbc.DefaultStatement.execute(DefaultStatement.java:102)
	at org.jooq.impl.SettingsEnabledPreparedStatement.execute(SettingsEnabledPreparedStatement.java:227)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:458)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:348)
	... 32 more
Caused by: com.simba.googlebigquery.googlebigquery.client.exceptions.JobExecutionErrorException: [Simba][BigQueryJDBCDriver](100032) Error executing query job. Message: BIGQUERY_API_ERR
	... 44 more
Caused by: com.simba.googlebigquery.support.exceptions.GeneralException: BIGQUERY_API_ERR
	at com.simba.googlebigquery.googlebigquery.client.requests.AbstractRequestWithRetry.shouldRetry(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.requests.jobs.JobsQueryRequest.shouldRetry(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.requests.AbstractRequestWithRetry.executeWithRetry(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.queryclient.JobsQueryClient.executeQuery(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.client.BQClient.executeQuery(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.dataengine.BQAbstractExecutor.execute(Unknown Source)
	at com.simba.googlebigquery.googlebigquery.dataengine.BQSQLExecutor.execute(Unknown Source)
	at com.simba.googlebigquery.jdbc.common.SStatement.executeNoParams(Unknown Source)
	at com.simba.googlebigquery.jdbc.common.BaseStatement.execute(Unknown Source)
	at org.jooq.tools.jdbc.DefaultStatement.execute(DefaultStatement.java:102)
	at org.jooq.impl.SettingsEnabledPreparedStatement.execute(SettingsEnabledPreparedStatement.java:227)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:458)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:348)
	at org.jooq.test.all.testcases.DDLTests.testCreateTableAsSelect(DDLTests.java:5475)
	at org.jooq.test.jOOQAbstractTest.testCreateTableAsSelect(jOOQAbstractTest.java:4718)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:757)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
```

**Comment by lukaseder** [comment]
A related feature that isn't supported in jOOQ in general, yet:

- https://github.com/jOOQ/jOOQ/issues/17025

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.20.0
